tornado/auth.py

```diff
@@ -985,9 +986,19 @@ def _on_access_token(self, redirect_uri, client_id, client_secret,
             future.set_exception(AuthError('Facebook auth error: %s' % str(response)))
             return
 
-        args = urlparse.parse_qs(escape.native_str(response.body))
+        try:
```

Comment:
Is there any way to tell which version we're dealing with instead of a try/catch? Do they set the content-type header correctly?

And it's a server-side API, so do we even need to deal with the older version, or did the new version just replace it? It looks like version 2.3 reached its [end-of-life today](https://developers.facebook.com/docs/apps/changelog).